comp/netflow/server/server.go

```diff
@@ -23,17 +23,21 @@ import (
 
 type dependencies struct {
 	fx.In
-	Config    nfconfig.Component
-	Logger    log.Component
-	Sender    sender.Component
```

Comment:
If in a future test we want to inject a mock sender, is there a pattern for that?

Reply:
Because there are several instances of `Sender` (the default sender and one per check), it is unclear if it should be a component as components are singletons when using fx. With a proper implementation of a [mock demultiplexer](https://github.com/DataDog/datadog-agent/pull/20020#discussion_r1355511917) it should be doable anyway.